src/main/java/storage/Storage.java

```diff
@@ -41,10 +46,14 @@ public static void createNewFile(Path path) throws IOException {
     public static void saveTasksToFile(Map<LocalDate, List<String>> tasks, Path path) {
         try (FileWriter writer = new FileWriter(path.toFile())) {
             for (Map.Entry<LocalDate, List<String>> entry : tasks.entrySet()) {
+                assert entry != null;
                 LocalDate date = entry.getKey();
+                assert date != null;
                 List<String> taskList = entry.getValue();
+                assert taskList != null;
```

Comment:
Good use of assertions here! You can consider adding strings at the end to describe the error if the assertion is not met. Referencing this in case you need it: https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/errorHandling.html#assertions